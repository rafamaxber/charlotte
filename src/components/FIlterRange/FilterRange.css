.rheostat {
  overflow: visible;
}

.rheostat-background {
  background-color: #eff1f1;
  position: relative;
  height: 4px;
}

.rheostat-progress {
  background-color: #f98339;
  position: absolute;
  top: 0;
  height: 4px;
}

.rheostat-handle {
  background-color: #fff;
  border: 4px solid #F98100;
  border-radius: 50%;
  height: 24px;
  outline: none;
  z-index: 2;
  width: 24px;
  top: -9px;
}
