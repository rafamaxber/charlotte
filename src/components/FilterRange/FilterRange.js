import React, { Component } from 'react';
import styled from 'styled-components';
import Rheostat from 'rheostat';
import './FilterRange.css';

const WrapperFilter = styled.div`
  max-width: 300px;
`;
const FilterTitle = styled.div`
  font-family: 'Montserrat', sans-serif;
  font-style: normal;
  font-weight: normal;
  line-height: normal;
  font-size: 22px;
  letter-spacing: 2.5px;
  color: #000000;
  margin-bottom: 39px;
`;
const TextSmall = styled.div`
  font-family: 'Heebo', sans-serif;
  font-style: normal;
  font-weight: normal;
  line-height: normal;
  font-size: 12.8px;
  color: #B5B5B5;
  margin-bottom: 20px;
`;
const FilterRange = styled.div``;
const WrapperValue = styled.div`
  display: flex;
  justify-content: space-between;
  margin-top: 20px;
`;
const FilterValue = styled.div`
  text-align: ${props => props.left ? 'left' : 'right' };
`;
const FilterValueTitle = styled.div`
  font-family: 'Heebo', sans-serif;
  font-style: normal;
  font-weight: normal;
  line-height: normal;
  font-size: 14px;
  color: #B5B5B5;
  margin-bottom: 5px;
`;
const FilterValueTextPrice = styled.div`
  font-family: 'Heebo', sans-serif;
  font-style: normal;
  font-weight: bold;
  line-height: normal;
  font-size: 22px;
  letter-spacing: 2.5px;
  color: rgba(249, 129, 0, 0.467816);
`;

export default class WrapperFilterRange extends Component {
  constructor(props) {
    super(props);
    this.state = this.getInitialState();
  }

  getInitialState() {
    return {
      min: 100,
      max: 1000,
      minPrice: 100,
      maxPrice: 930,
    }
  }

  updateFilterValues() {
    this.props.updateFilterPrice({ minPrice: this.state.minPrice, maxPrice: this.state.maxPrice });
    return this.props.afterupdateFilterPrice();
  }

  render() {
    return(
      <WrapperFilter>
        <FilterTitle>
          Filters
        </FilterTitle>
        <FilterRange>
          <TextSmall>
            Price Range
          </TextSmall>
          <Rheostat
            min={this.state.min}
            max={this.state.max}
            onValuesUpdated={({ values }) => this.setState({ minPrice: values[0], maxPrice: values[1] })}
            onSliderDragEnd={this.updateFilterValues.bind(this)}
            values={[this.state.minPrice, this.state.maxPrice]}
          />
          <WrapperValue>
            <FilterValue left>
              <FilterValueTitle>Min</FilterValueTitle>
              <FilterValueTextPrice>
                ${this.state.minPrice}
              </FilterValueTextPrice>
            </FilterValue>

            <FilterValue right>
              <FilterValueTitle>Max</FilterValueTitle>
              <FilterValueTextPrice>
                ${this.state.maxPrice}
            </FilterValueTextPrice>
            </FilterValue>

          </WrapperValue>
        </FilterRange>
      </WrapperFilter>
    );
  }
}
