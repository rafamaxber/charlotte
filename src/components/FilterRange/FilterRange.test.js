import React from 'react';
import renderer from 'react-test-renderer';
import FilterRange from './FilterRange';

describe('FilterRange', () => {
  it('Should match with snapshot', () => {
    const component = renderer.create(
      <FilterRange></FilterRange>
    );
    expect(component).toMatchSnapshot();
  });
});
