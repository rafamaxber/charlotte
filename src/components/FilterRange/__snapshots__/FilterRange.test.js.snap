// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterRange Should match with snapshot 1`] = `
<div
  className="sc-bdVaJa tsCoj"
>
  <div
    className="sc-htpNat KtFsv"
  >
    <div
      className="sc-bwzfXH iGTVFs"
    >
      Price Range
    </div>
    <div
      className="rheostat rheostat-horizontal "
      onClick={[Function]}
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <div
        className="rheostat-background"
      />
      <button
        aria-disabled={false}
        aria-valuemax={1000}
        aria-valuemin={100}
        aria-valuenow={undefined}
        className="rheostat-handle"
        data-handle-key={0}
        onClick={undefined}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        role="slider"
        style={
          Object {
            "left": "NaN%",
            "position": "absolute",
          }
        }
        tabIndex={0}
        type="button"
      />
      <button
        aria-disabled={false}
        aria-valuemax={1000}
        aria-valuemin={100}
        aria-valuenow={undefined}
        className="rheostat-handle"
        data-handle-key={1}
        onClick={undefined}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        role="slider"
        style={
          Object {
            "left": "NaN%",
            "position": "absolute",
          }
        }
        tabIndex={0}
        type="button"
      />
      <div
        className="rheostat-progress"
        style={
          Object {
            "left": "NaN%",
            "width": "NaN%",
          }
        }
      />
    </div>
    <div
      className="sc-bxivhb bCEIsr"
    >
      <div
        className="sc-ifAKCX TiiUY"
      >
        <div
          className="sc-EHOje jAnwVP"
        >
          Min
        </div>
        <div
          className="sc-bZQynM klfGCS"
        >
          $
        </div>
      </div>
      <div
        className="sc-ifAKCX gENqNG"
      >
        <div
          className="sc-EHOje jAnwVP"
        >
          Max
        </div>
        <div
          className="sc-bZQynM klfGCS"
        >
          $
        </div>
      </div>
    </div>
  </div>
</div>
`;
